copy/copy.go

```diff
@@ -158,9 +158,12 @@ func Image(policyContext *signature.PolicyContext, destRef, srcRef types.ImageRe
 		}
 	}()
 
-	if err := checkImageDestinationForCurrentRuntimeOS(src, dest); err != nil {
-		return err
-	}
+	// Disabling checkImageDestinationForCurrentRuntimeOS since the implementation is basing decision on whether there is a match on the current OS
```

Comment:
@mtrmac I commented this check out. I think the current implementation is not correct and w/o the ability to control this behaviour in some way, I don't think it is really useful. 

I don't think that the OS of the host can be used to determine whether a copy can occur. The destination of the copy might be remote. Or am I missing something?

Reply:
`dest` _already_ can choose whether this check make sense, via `ImageDestination.MustMatchRuntimeOS`; and this check is actually useful for copying to local `docker-daemon:`.

Maybe a specific transport should change its value of `MustMatchRuntimeOS`? Perhaps `docker-daemon:` _when connecting to a remote host_?

Reply:
> dest already can choose whether this check make sense, 

Right, but it is hardcoded. In the case of the Docker daemon  _daemon_dest.go_. Given that a Docker daemon can also be remote, having _MustMatchRuntimeOS()_ always return true is preventing me to copy an image from a Windows host to a Docker daemon, even though it should be possible.

> and this check is actually useful for copying to local docker-daemon:.

Maybe. 

> Maybe a specific transport should change its value of MustMatchRuntimeOS? Perhaps docker-daemon: when connecting to a remote host?

Right. So one solution would be to me _daemonImageDestination.MustMatchRuntimeOS()_ stateful. 
In the simplest case one could just use _dockerclient.DefaultDockerHost_. If the default is used, we return _true_, if the SystemContext explicitly specifies a value for _DockerDaemonHost_ we return false (unless it maybe matches `dockerclient.DefaultDockerHost`). What I want to avoid is to make any decisions based on the IP/host specified in _ctx.DockerDaemonHost_. Maybe something like 127.0.0.1 or localhost would still make sense, but everything else is not easily verifiable.  

WDYT? Make _daemonImageDestination.MustMatchRuntimeOS()_ stateful?

Reply:
@mtrmac Any feedback regarding the other changes in this pull request? Are we good on these?

Reply:
@mtrmac I made _MustMatchRuntimeOS()_ conditional. WDYT?

Reply:
> @mtrmac I made MustMatchRuntimeOS() conditional. WDYT?

Looks good. As you say, some of the cases are difficult to determine, and most users shouldn’t hit them. If someone does hits them,. we might need to make the tests a bit more precise and change the semantics; hopefully that won’t affect any previously working users.